src/App.tsx

```diff
@@ -1,24 +1,33 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import { Route, Routes, useLocation } from "react-router-dom";
+import "./App.css";
+import Nav from "./components/Layout/Nav";
+import Dashboard from "./components/Pages/Dashboard/Dashboard";
+import HabitForm from "./components/Pages/Form/Form";
+import Knowledge from "./components/Pages/Knowledge/Knowledge";
+import Profile from "./components/Pages/Profile/Profile";
+import Tracker from "./components/Pages/Tracker/Tracker";
+
+import Welcome from "./components/Pages/Welcome/Welcome";
 
 function App() {
+  const location = useLocation();
+  const savedHabit = localStorage.getItem("habit");
```

Comment:
I'd move all related places to this storage to one file (ideally one class or a few utility functions). Alternatively you can export constants to the storage keys - e.g. `"habit"` and then use it still with `localStorage.getItem()` but then it may be a bit harder to change the underlying storage provider in the future.